apps/aecore/src/aec_headers.erl

```diff
@@ -648,3 +667,22 @@ validate_max_time({Header, _}) ->
         false ->
             {error, block_from_the_future}
     end.
+
+%% TODO: this is taken from ws_int_dispatch.erl - needs refactoring after
+%% separation of key/micro block records.
```

Comment:
They are separated already?!

Reply:
There seems to be more work done to separate them. Both block types use the same db table. When reading a block, we find out what type it is from the [returned value](https://github.com/aeternity/epoch/blob/master/apps/aecore/src/aec_db.erl#L237).